site/content/contribute/gsoc/_index.md

```diff
@@ -0,0 +1,223 @@
+-   **Skills required**: React, Go
+-   **Difficulty**: Hard
+-   **Mentors**: Daniel Espino García, Alejandro García Montoro
+
+## Plugins
+
+Plugins are the de facto way of extending Mattermost without modifying the product itself: They are web applications on their own, whose server part directly talks to the Mattermost server, while the web part is injected in the Mattermost web application.
+
+Although some of them, as the Quote plugin, will require a change in the plugin infrastructure, these ideas are usually more flexible, as they are not direct modifications to the product itself, but independent pieces of software that users can decide to install and enable.
+
+### Spotify plugin
+
+There are tons of integrations that are potentially useful for our users. One that seems to be asked for quite a lot is an integration with Spotify. There are several features that we would like to see in such a plugin, but the development of this idea is open to the interests of the student:
+
+-   Share what the user is listening to in a pre-defined channel.
+-   Add a new section to the profile popover that shows the current song being played.
+-   Add player controls inside Mattermost: Location to be decided, but we would love to control what we are listening to without ever leaving Mattermost.
+-   Collective playlist with all the songs shared to a channel.
+
+This project is open to ideas that should be reflected in the student proposal.
+
+-   **Expected outcomes**: A new plugin to integrate Spotify with Mattermost.
+-   **Skills required**: React, Go
+-   **Difficulty**: Medium
+-   **Mentors**: Alejandro García Montoro
+
+### Quote plugin
+
+When we are writing in a busy thread, and we want to answer a specific post, there's an unwritten rule that says that you should prepend your post with a quote of that message, formatting it as a quote with the usual Markdown syntax. This is a manual process, and as such, it is sometimes skipped. But what if we could automate it?
+
+This project has that simple purpose: implement a new plugin that would add a new entry, `Quote`, to the post menu that would copy the corresponding post in the input box, properly formatted by prepending it with `> `. The plugin should never delete the current text in the input box, only adding this block of text.
+
+However, as of today, plugins are not able to modify the input textbox, so the first and most important task in this project would be to change the plugin API to add such a feature. These architectural changes are valuable in themselves, so a project that would only deliver this would be considered successful. However, if the student finishes this task before the coding period ends, the next task would be to actually implement the quote plugin, using the new functionality as a basis for it.
```

Comment:
```suggestion
Currently, plugins aren't able to modify the input box, so the first and most important task in this project would be to change the plugin API to support such a feature. These architectural changes are valuable in themselves, so a project that would only deliver this would be considered successful. However, if the student finishes this task before the coding period ends, the next task would be to actually implement the quote plugin, using the new functionality as a basis for it.
```